src/index.d.ts

```diff
@@ -43,13 +43,14 @@ export type Ref<T> = RefObject<T> | RefCallback<T> | null;
 
 export type ComponentChild =
 	| VNode<any>
-	| object
 	| string
 	| number
+	| Array<ComponentChild>
 	| bigint
 	| boolean
 	| null
-	| undefined;
+	| undefined
+	| Function;
 export type ComponentChildren = ComponentChild[] | ComponentChild;
```

Comment:
I wonder, do we really need to alter the type here? I do generally prefer `ComponentChild` being singular, with `ComponentChildren` being used if an array of children is needed.

If it's a compat issue, we can simply alter the [`ReactNode`](https://github.com/preactjs/preact/blob/f7f9d9bf4309d09bcd0a432acd72787b7d9baecb/compat/src/index.d.ts#L66) type instead?